eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/RelationConverter.php

```diff
@@ -146,7 +150,10 @@ public function toFieldDefinition(StorageFieldDefinition $storageDef, FieldDefin
             return;
         }
 
-        if ($selectionType = $dom->getElementsByTagName('selection_type')) {
+        if (
+            ($selectionType = $dom->getElementsByTagName('selection_type')) &&
+            $selectionType->item(0)->hasAttribute('value')
+        ) {
             $fieldSettings['selectionMethod'] = (int)$selectionType->item(0)->getAttribute('value');
```

Comment:
Quick test showed me that `getElementsByTagName` never returns an empty value. It's always `\DOMNodeList`. It has property `length`. If `length = 0`, then `$selectionType->item(0)` will return `null` and calling `hasAttribute` will throw an exception.
So rather:
`if (($selectionTypeFirstItem = $dom->getElementsByTagName('selection_type')->item(0)) && $selectionTypeFirstItem->hasAttribute('value'))`

Reply:
Ok, so DomList never evaluates to empty here?

Reply:
Yes. Consider the following example (just re-pasting from elsewhere to keep discussion here consistent):
```php
$xml = <<< XML
<?xml version="1.0" encoding="utf-8"?>
<para></para>
XML;

$dom = new DOMDocument;
$dom->loadXML($xml);
if (($elements = $dom->getElementsByTagName('non_existent_tag'))) {
	var_dump($elements->item(0));
}
```
will output `NULL`.